### DISCUSSION TITLE
How to create a gradient at the image border to blend into the background

### DISCUSSION THREAD

**Discussion by DougBoffey** [ImageMagick/ImageMagick#3794] [Digital Image Processing]
How can I alter the border of a picture, so it gradually blends in with a background colour?

**Comment by snibgo** [comment]
What version of IM, on what platform?

Please show an example of the effect you want. Ideally, include inputs as well as the output.

**Reply by DougBoffey** [reply]
ImageMagick 6.9.10-23 Q16 x86_64 20190101 running on Ubuntu

As for an example of what I want to achieve, I don’t have such an example.

What I want is to have the the interior of the image remain, up to within 10 (or whatever) pixels of each edge, with the edge 10 pixels gradually changing to a background colour, on which the image is to be displayed, so that there is no hard border around the image.

**Comment by fmw42** [comment]
Here is one way (Unix syntax). 

```
Read the input
Create the color image
Create a black image and blur the outside with white to make a mask
Composite the 3 images using the mask to blend the first two.

```
Input:

![lena](https://user-images.githubusercontent.com/13458218/121048978-253bbf00-c76c-11eb-9a45-1c7cf5af4431.png)

```
convert lena.png \
\( -clone 0 -fill blue -colorize 100 \) \
\( -clone 0 -fill black -colorize 100 \
-virtual-pixel white -blur 0x30 \) \
-compose over -composite result.png

```

![result](https://user-images.githubusercontent.com/13458218/121049220-5d430200-c76c-11eb-9acf-192f207b370f.png)
